### PR TITLE
[ISSUE #570]ASoC connect runtime optimization: CLI

### PR DESCRIPTION
**Pull request by Dreaouth**
## What is the purpose of the change

XXXXX

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

File: rocketmq-connect/rocketmq-connect-runtime/src/main/java/org/apache/rocketmq/connect/runtime/rest/RestSender.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package org.apache.rocketmq.connect.runtime.rest;
```
Comment (imaffe) [inline rocketmq-connect/rocketmq-connect-runtime/src/main/java/org/apache/rocketmq/connect/runtime/rest/RestSender.java:1]: We need to add Apache header at the start of each file~

File: rocketmq-connect/rocketmq-connect-runtime/src/main/java/org/apache/rocketmq/connect/runtime/tools/command/StopAllSubCommand.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    @Override
+    public Options buildCommandlineOptions(Options options) {
+        return options;
+    }
+
+    @Override
+    public void execute(CommandLine commandLine, Options options) throws SubCommandException {
+        try {
+            String url = "http://localhost:8081/connectors/" + commandName();
```
Comment (imaffe) [inline rocketmq-connect/rocketmq-connect-runtime/src/main/java/org/apache/rocketmq/connect/runtime/tools/command/StopAllSubCommand.java:27]: Why this url is hardcoded in code ~

File: rocketmq-connect/rocketmq-connect-runtime/src/main/java/org/apache/rocketmq/connect/runtime/tools/command/CreateConnectorSubCommand.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    @Override
+    public void execute(CommandLine commandLine, Options options) throws SubCommandException {
+        try{
+            String connectorName = commandLine.getOptionValue('c').trim();
+            String baseUrl = "http://localhost:8081/connectors/" + connectorName + "?config=";
+            String filePath = commandLine.getOptionValue('p').trim();
+            String config = readFile(filePath);
+            System.out.println("Send request to " + baseUrl + config);
```
Comment (imaffe) [inline rocketmq-connect/rocketmq-connect-runtime/src/main/java/org/apache/rocketmq/connect/runtime/tools/command/CreateConnectorSubCommand.java:41]: Is this format clear enough for users?

File: rocketmq-connect/rocketmq-connect-runtime/src/main/java/org/apache/rocketmq/connect/runtime/tools/MQConnectAdminStartup.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MQConnectAdminStartup {
+    protected static List<SubCommand> subCommandList = new ArrayList<SubCommand>();
+
+
+    public static void main(String[] args) {
```
Comment (imaffe) [inline rocketmq-connect/rocketmq-connect-runtime/src/main/java/org/apache/rocketmq/connect/runtime/tools/MQConnectAdminStartup.java:20]: how about lets add a readme for this CLI tool ~

File: rocketmq-connect/rocketmq-connect-runtime/src/main/java/org/apache/rocketmq/connect/runtime/rest/RestSender.java (hunk 1 above)
Comment (Dreaouth) [reply rocketmq-connect/rocketmq-connect-runtime/src/main/java/org/apache/rocketmq/connect/runtime/rest/RestSender.java]: I have added Apache header at the start of each file

File: rocketmq-connect/rocketmq-connect-runtime/src/main/java/org/apache/rocketmq/connect/runtime/tools/command/StopAllSubCommand.java (hunk 2 above)
Comment (Dreaouth) [reply rocketmq-connect/rocketmq-connect-runtime/src/main/java/org/apache/rocketmq/connect/runtime/tools/command/StopAllSubCommand.java]: I have avoid hardcoded by adding the configuration file

File: rocketmq-connect/rocketmq-connect-runtime/src/main/java/org/apache/rocketmq/connect/runtime/tools/MQConnectAdminStartup.java (hunk 4 above)
Comment (Dreaouth) [reply rocketmq-connect/rocketmq-connect-runtime/src/main/java/org/apache/rocketmq/connect/runtime/tools/MQConnectAdminStartup.java]: Added README

**Pull request closed without merging**